### PR TITLE
better docs for compiler - related #98

### PR DESCRIPTION
**Pull request by jamesward**

### CODE REVIEW

File: compiler/README.md [hunk 1]
```diff
@@ -1,55 +1,156 @@
...
 `.proto` file. It works with the Protobuf Compiler (`protoc`).
 
-Normally you don't need to compile the codegen by yourself, since pre-compiled
-binaries for common platforms are available on Maven Central:
+### Build Tool Plugins
 
-1. Navigate to https://mvnrepository.com/artifact/io.grpc/protoc-gen-grpc-kotlin
-2. Select a package version
-3. Click "Files: View All"
+Usually this compiler is used via a built took plugin, like in Gradle, Maven, etc.
```
Comment (bshaffer) [inline compiler/README.md:9]: Should be `build tool plugin`?

File: compiler/README.md [hunk 2]
```diff
@@ -1,55 +1,156 @@
...
 
-However, if the pre-compiled binaries are not compatible with your system,
-you may want to build your own codegen.
+For Gradle include the [protobuf plugin](https://github.com/google/protobuf-gradle-plugin) with at least version `0.8.13`, like:
```
Comment (bshaffer) [inline compiler/README.md:11]: nit: add comma: `For Gradle, include...`

File: compiler/README.md [hunk 3]
```diff
@@ -1,55 +1,156 @@
...
+            it.plugins {
+                id("grpc")
+                id("grpckt")
+            }
+        }
+    }
+}
+```
+
+For Maven include the [protobuf plugin](https://www.xolstice.org/protobuf-maven-plugin/) and configure it to use the
```
Comment (bshaffer) [inline compiler/README.md:50]: Same here, `For Maven, include...`

File: compiler/README.md (hunk 1 above)
Comment (jamesward) [reply compiler/README.md:9]:
```suggestion
Usually this compiler is used via a build tool plugin, like in Gradle, Maven, etc.
```

File: compiler/README.md (hunk 2 above)
Comment (jamesward) [reply compiler/README.md:11]:
```suggestion
For Gradle, include the [protobuf plugin](https://github.com/google/protobuf-gradle-plugin) with at least version `0.8.13`, like:
```

File: compiler/README.md (hunk 3 above)
Comment (jamesward) [reply compiler/README.md:50]:
```suggestion
For Maven, include the [protobuf plugin](https://www.xolstice.org/protobuf-maven-plugin/) and configure it to use the
```

**Review by bshaffer — APPROVED** [review]

**Pull request merged**